lib/LocalTopicModel.js

```diff
@@ -0,0 +1,464 @@
+var LOWER_END = 122;
+var ASCII_END = 127;
+
+var MAXIMUM_TERM_LENGTH = 30;
+var UPDATES = 64;
+
+var CODE_TO_NAME = {
+    "da": "danish",
+    "nl": "dutch",
+    "en": "english",
+    "fi": "finnish",
+    "fr": "french",
+    "de": "german",
+    "hu": "hungarian",
+    "it": "italian",
+    "nn": "norwegian",
+    "pt": "portuguese",
+    "ro": "romanian",
+    "ru": "russian",
+    "es": "spanish",
+    "sv": "swedish",
+    "tr": "turkish"
+}
+
+if (NODEJS) {
+  var util = require('util');
+  var events = require('events');
+  var TopicModel = require(PATH + 'TopicModel');
+  var Snowball = require('snowball');
+  var MersenneTwister = require('mersenne-twister');
+}
+var utils = require(PATH + 'utils');
+var constants = require(PATH + 'constants');
+
+function isalnum(astr, idx) {
+  var code = astr.charCodeAt(idx);
+
+  if (code >= ZERO_CODE && code <= LOWER_END) {
+    if (code <= NINE_CODE) return true;
+    if (code >= UPPER_BEGIN && code <= UPPER_END) return true;
+    if (code <= LOWER_END) return true;
+    return false;
+  } else if (code >= ASCII_END) {
+    // This will break for CJK / Arabic alphabets, but so will everything else.
+    if (astr.charAt(idx).toLowerCase() != astr.charAt(idx).toUpperCase())
+      return true;
+    else return false;
+  } else return false;
+}
+
+/**
+ * LocalTopicModel: Simplified local object for the topicmodel resource.
+ * @constructor
+ */
+function LocalTopicModel(resource, connection) {
+  /**
+   * Constructor for the LocalTopicModel local object.
+   *
+   * @param {object} resource BigML topicmodel resource
+   * @param {object} connection BigML connection
+   */
+
+  var self, topicModel, fillStructure, language, i, tstem;
+
+  this.resourceId = utils.getResource(resource);
+  if ((typeof this.resourceId) === 'undefined') {
+    throw new Error('Cannot build a TopicModel from this resource: ' + resource);
+  }
+
+  this.fields = null;
+  this.ready = null;
+  this.stemmer = null;
+  this.ready = null;
+  this.seed = null;
+  this.caseSensitive = null;
+  this.bigrams = null;
+  this.ntopics = null;
+  this.temp = null;
+  this.uniformDocAssignments = null;
+  this.uniformNormalizer = null;
+  this.phi = null;
+  this.termToIndex = null;
+
+  self = this;
+
+  fillStructure = function (error, resource) {
+    /**
+     * Auxiliary function to load the resource info in the TopicModel structure.
+     *
+     * @param {object} error Error info
+     * @param {object} resource Model's resource info
+     */
+    var model, status, fields, index, fieldId, field, assignments, beta,
+        sums, nterms, norm, i, j, k, w;
+
+    if (error) {
+      throw new Error('Cannot create the TopicModel instance. Could not' +
+                      ' retrieve the resource: ' + error);
+
+    }
+
+    status = utils.getStatus(resource);
+    if ((typeof resource.object) !== 'undefined') {
+      resource = resource.object;
+    }
+    if ((typeof resource.model) !== 'undefined') {
```

Comment:
Yes, it's topic_model now. It's the first error that the module gave me and I was gonna report it right now.